lib/ansible/plugins/lookup/first_found.py

```diff
@@ -241,8 +241,8 @@ def run(self, terms: list, variables=None, **kwargs):
             subdir += "s"  # convert to the matching directory name
         else:
             terms = _recurse_terms(terms, omit_undefined=False)  # undefined values are only omitted when invoked using `with`
-
-            subdir = None
+            # set default value to 'files' to look under files directory in the role
+            subdir = "files"
```

Comment:
the subdir depend son type of task, for example an include_vars would expect a 'vars' directory, template expects 'templates' to be the subdir.